File: csharp/GetCryptoBookBids.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
  public class GetCryptoBookBids
  {
    public static void Main()
    {
      Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

      var cryptoApi = new CryptoApi();
      var pair = "btcusd";  // string | Return the order book bids for the given Crypto Currency Pair. (optional) 
      var exchange = "gemini";  // string | Return the order book bids for a Crypto Currency on the given Crypto Exchange. (optional) 
      var currency = "BTC";  // string | Return the order book bids for the given Crypto Currency. (optional) 

      try
      {
        ApiResponseCryptoBookBids result = cryptoApi.GetCryptoBookBids(pair, exchange, currency);
        List<CryptoBid> crypto_bids = result.Bids;

        CryptoPairSummary crypto_pair = result.Pair;
        Console.WriteLine("Crypto Currency Pair: " + crypto_pair.Name);

        CryptoExchangeSummary crypto_exchange = result.Exchange;
        Console.WriteLine("Crypto Exchange: " + crypto_exchange.Name);

        Console.WriteLine();
        Console.WriteLine("----------------- BIDS -----------------");

        crypto_bids.ForEach(delegate (CryptoBid bid)
        {
          Console.WriteLine();
          Console.WriteLine("Price: " + bid.Price);
          Console.WriteLine("Size:  " + bid.Size);
        });
      }
      catch (Exception e)
      {
        Debug.Print("Exception when calling CryptoApi.GetCryptoBookBids: " + e.Message);
      }
    }
  }
}

File: python/get_crypto_book_bids.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
crypto_api = intrinio_sdk.CryptoApi()
 
pair = 'btcusd' # str | Return the order book bids for the given Crypto Currency Pair. (optional)
exchange = 'gemini' # str | Return the order book bids for a Crypto Currency on the given Crypto Exchange. (optional)
currency = 'BTC' # str | Return the order book bids for the given Crypto Currency. (optional)
 
try:
    api_response = crypto_api.get_crypto_book_bids(pair=pair, exchange=exchange, currency=currency)
    bids = api_response.bids

    print("Crypto Currency Pair: %s" % (api_response.pair.name))
    print("Crypto Exchange: %s" % (api_response.exchange.name))
    print("")
    print("-------------------------------------- BIDS -------------------------------------- ")

    for bid in bids:
        print("")
        print("Price: %s" % (bid.price))
        print("Size:  %s" % (bid.size))

except ApiException as e:
    print("Exception when calling CryptoApi->get_crypto_book_bids: %s\n" % e)
